In [52]:
import pandas as pd
import seaborn as sns
import matplotlib as plt
import numpy as np

loandata=pd.read_csv('loan.csv')
loandata.head()


loanprep=loandata.copy()
loanprep.dropna(inplace=True)

# dropping data which are not required



loanprep=loanprep.drop(['gender'],axis=1)
# axis=0 refers to rows.
# axis=1 refers to columns.


loanprep=pd.get_dummies(loanprep,drop_first=True)



# for normalising data
from sklearn.preprocessing import StandardScaler
scalar=StandardScaler()

loanprep['income']=scalar.fit_transform(loanprep[['income']])
loanprep['loanamt']=scalar.fit_transform(loanprep[['loanamt']])
# means ye normal distribution me chala jaega mean=0,stddev=1



y=loanprep[['status_Y']]
x=loanprep.drop(['status_Y'],axis=1)

from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=1234,stratify=y)


from sklearn.linear_model import LogisticRegression

lr=LogisticRegression()
lr.fit(x_train,y_train)

C:\Users\sutar\Contacts\sGsgW\Lib\site-packages\sklearn\utils\validation.py:1143: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


LogisticRegression()

In [33]:
prediction=lr.predict(x_train)


In [16]:
from sklearn.metrics import confusion_matrix
cm=confusion_matrix(y_test,ps)

In [9]:
ps=lr.predict(x_test)

In [17]:
cm

array([[ 29,  20],
       [  2, 108]], dtype=int64)

# from that we correcty approved 108 and corectly rejected 29 but we have approved 20 instead of rejecting same vice versa for 2


In [18]:
# out of 159 we correctly predicted 138 of them

In [25]:
score=lr.score(x_test,y_test)
score# this is accuracy

0.8616352201257862

In [27]:
loanprep

ch    income   loanamt  married_Yes  status_Y
1    1.0 -0.128073 -0.194250         True     False
2    1.0 -0.392077 -0.971015         True      True
3    1.0 -0.461621 -0.294478         True      True
4    1.0  0.108246 -0.031380        False      True
5    1.0  0.011017  1.547205         True      True
..   ...       ...       ...          ...       ...
609  1.0 -0.408754 -0.908372        False      True
610  1.0 -0.207624 -1.296754         True      True
611  1.0  0.453802  1.371807         True      True
612  1.0  0.372249  0.544929         True      True
613  0.0 -0.128073 -0.131608        False     False

[529 rows x 5 columns]

In [46]:
input_data=(0.0,-0.13,-0.17,True)
new_data=np.asarray(input_data).reshape(1,-1)


In [47]:
lr.predict(new_data) 

C:\Users\sutar\Contacts\sGsgW\Lib\site-packages\sklearn\base.py:439: UserWarning: X does not have valid feature names, but LogisticRegression was fitted with feature names
  warnings.warn(


array([False])

In [51]:
a=input(float())
b=input(float())
c=in

0.0 54564652.656454


In [53]:
loanprep

ch    income   loanamt  married_Yes  status_Y
1    1.0 -0.128073 -0.194250         True     False
2    1.0 -0.392077 -0.971015         True      True
3    1.0 -0.461621 -0.294478         True      True
4    1.0  0.108246 -0.031380        False      True
5    1.0  0.011017  1.547205         True      True
..   ...       ...       ...          ...       ...
609  1.0 -0.408754 -0.908372        False      True
610  1.0 -0.207624 -1.296754         True      True
611  1.0  0.453802  1.371807         True      True
612  1.0  0.372249  0.544929         True      True
613  0.0 -0.128073 -0.131608        False     False

[529 rows x 5 columns]

tion errors.